cli/list.go

```diff
@@ -112,7 +112,7 @@ func (r *RootCmd) list() *serpent.Command {
 				return err
 			}
 
-			if len(res) == 0 {
+			if len(res) == 0 && formatter.FormatID() != cliui.JSONFormat().ID() {
 				pretty.Fprintf(inv.Stderr, cliui.DefaultStyles.Prompt, "No workspaces found! Create one:\n")
 				_, _ = fmt.Fprintln(inv.Stderr)
 				_, _ = fmt.Fprintln(inv.Stderr, "  "+pretty.Sprint(cliui.DefaultStyles.Code, "coder create <name>"))
```

Comment:
Unfortunate we print anything outside the formatter. Since this code is not going to get more formatters anytime soon, this works for me :+1: 

---

There might be value in some formatter wrapping like we do here: https://github.com/coder/coder/blob/a12f553130a56ec2add6f7b7786b38a642844770/cli/cliui/output.go#L217-L217

Something to switch the formatter based on the output or something :thinking: